From 5a2bc1357044734330ac78e05f5ab4ec8b186772 Mon Sep 17 00:00:00 2001
From: Ginu Jacob <ginu.jacob@arm.com>
Date: Thu, 11 Sep 2025 14:43:07 +0100
Subject: [PATCH 6/6] VK_EXT_frame_boundary: Fixing the sType used in
 VkFrameBoundaryEXT

In this change, the sType used in VkFrameBoundaryEXT instance is fixed.

Upstream-Status: Backport [5a2bc1357044734330ac78e05f5ab4ec8b186772]
Signed-off-by: Ginu Jacob <ginu.jacob@arm.com>
Change-Id: Id1dab87b9b4144d6030c57dbe4e33a41e7045a9f
---
 wsi/extensions/frame_boundary.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wsi/extensions/frame_boundary.cpp b/wsi/extensions/frame_boundary.cpp
index 4617d25..453bdc7 100644
--- a/wsi/extensions/frame_boundary.cpp
+++ b/wsi/extensions/frame_boundary.cpp
@@ -63,7 +63,7 @@ std::optional<VkFrameBoundaryEXT> wsi_ext_frame_boundary::handle_frame_boundary_
 std::optional<VkFrameBoundaryEXT> create_frame_boundary(const VkPresentInfoKHR &present_info)
 {
    auto *present_frame_boundary =
-      util::find_extension<VkFrameBoundaryEXT>(VK_STRUCTURE_TYPE_PRESENT_ID_KHR, present_info.pNext);
+      util::find_extension<VkFrameBoundaryEXT>(VK_STRUCTURE_TYPE_FRAME_BOUNDARY_EXT, present_info.pNext);
 
    /* Extract the VkFrameBoundaryEXT structure to avoid passing other, unrelated structures to vkQueueSubmit */
    if (present_frame_boundary != nullptr)
-- 
2.34.1

